clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -243,33 +243,20 @@ public MemoryRecords build() {
     /**
      * Get the max timestamp and its offset. The details of the offset returned are a bit subtle.
      *
-     * If the log append time is used, the offset will be the last offset unless no compression is used and
```

Comment:
@showuon @junrao thanks for this change and I learn a lot from it. 

BTW, should we document this behavior? I mean how we decide the offset of max timestamp when there are multi-records having same timestamp.

Reply:
Updated. Thanks.

Reply:
Could you please update `OffsetSpec` also? https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/OffsetSpec.java#L65